src/parseTools.js

```diff
@@ -1477,6 +1477,34 @@ function makeStaticString(string) {
   return '(stringToUTF8("' + string + '", ' + ptr + ', ' + len + '), ' + ptr + ')';
 }
 
+// Generates access to module exports variable in pthreads worker.js
+function makeAsmExportAccessInPthread(variable) {
+  if (MODULARIZE) {
+    return "Module['" + variable + "']" // 'Module' is defined in worker.js local scope, so not EXPORT_NAME in this case.
+  } else {
+    return EXPORT_NAME + "['" + variable + "']"
+  }
+}
+
+// Generates access to a global scope variable in pthreads worker.js
+function makeAsmGlobalAccessInPthread(variable) {
+  if (MODULARIZE) {
+    return "Module['" + variable + "']" // 'Module' is defined in worker.js local scope, so not EXPORT_NAME in this case.
+  } else {
+    return variable
+  }
+}
+
+// Generates access to both global scope variable and exported Module variable, e.g. "Module['foo'] = foo" or just plain "foo" depending on if we are MODULARIZEing.
+// Used the be able to initialize both variables at the same time.
+function makeAsmExportAndGlobalAccessInPthread(variable) {
```

Comment:
This is for an assignment target, I think? The name could specify that, and so it would not be odd that one code path has side effects and one seems to not (as in all cases this is the assign target, so there are side effects).

How about `makeAsmExportAndGlobalAssignTargetInPthread`?

Reply:
ok